py7zr/compression.py

```diff
@@ -102,6 +108,28 @@ def extract_single(self, fp: Union[BinaryIO, str], files, src_start: int, src_en
                 with NullIO() as ofp:
                     self.decompress(fp, f.folder, ofp, f.uncompressed[-1], f.compressed, src_end)
 
+    def extract_single_to_dict(self, fp: Union[BinaryIO, str], files, src_start: int, src_end: int) -> None:
```

Comment:
Here is many copy of code. Could you reduce code duplication for maintainability?

In version 0.5.x, we have a  way to return data as io.ByteIO() for unit test purpose, but it was dropped in v0.6.
https://github.com/miurahr/py7zr/blob/9342a5dd37ec39ec68d09e67cfb0fc0fbfa4eeb3/py7zr/compression.py#L70-L100

You can reuse this. 

In this way,  `self.target_filepath.get()` may return a object which has a memory buffer instead of file-object, and have a write(), read() functions like file-object.  then main logic itself is not necessary to modify.